satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -555,11 +564,14 @@ def __init__(self, filename, filename_info, filetype_info,  # noqa: D417
         self.mask_bad_quality = mask_bad_quality
         nc_raw = xr.open_dataset(
             filename,
-            chunks={"x": CHUNK_SIZE,
-                    "y": CHUNK_SIZE,
-                    "x_ir_wv": CHUNK_SIZE,
-                    "y_ir_wv": CHUNK_SIZE}
+            # chunks={"x": CHUNK_SIZE,
+            #         "y": CHUNK_SIZE,
+            #         "x_ir_wv": CHUNK_SIZE,
+            #         "y_ir_wv": CHUNK_SIZE},
```

Comment:
Thanks for the first comments!
Concerning the chunks, it throws a ValueError at the moment: "ValueError: This function cannot handle duplicate dimensions, but dimensions {'srf_size'} appear more than once on this object's dims: ('srf_size', 'srf_size')"

It could be related to this post: https://github.com/pydata/xarray/issues/8579

Any other suggestions than disabling it are very welcome :)

Reply:
In that discussion they propose a workaround:
```
ds.variables["covariance..."].dims = ("srf_size_1", "srf_size_2")
ds.chunk(mychunks)
```
I tried that here: https://github.com/sfinkens/satpy/commit/82eb8cd67c6d7119b1cabeb4b34683b1c1f8b47e Let me know what you think!

I also updated the tests to trigger that situation:
1) Add a new variable with identical dimension names
2) Write a fake file to disk
3) Read it back in

Reply:
There's still a bit of work to be done, because the tests now fail due to some extra attributes and coordinates.

Reply:
Thank you for your feedback! As already discussed, I will include your proposed changes. Concerning the tests, I am currently working on it.

Reply:
@sfinkens I submitted some changes and the tests are mostly ok now. The test "test_reassign_coords(self)" still fails - as I am not familiar with the Mock Object Library, it would be great if you could have a look on it - thank you!

Reply:
Ok, I'll check after lunch!

Reply:
sfinkens and I did some pair programming: Calling time.astype("datetime64[s]").astype("datetime64[ns]") is not properly working because a float input is needed. We propose to open the dataset with decode_cs=False and afterwards decode time and other variables separatly to properly take care of time FillValues and offsets.

sfinkens added a test for the Interpolator.

The test "test_reassign_coords() is still failing because the new functionalities in DatasetWrapper() are not yet considered (mocking should not be of importance here). Maybe we should call the assign_coords() method directly instead of DatasetWrapper().

Also, a test for DatasetWrapper should be included to test the separate encoding.

Reply:
Test is working again, DatasetWrapper is implicitly tested within the other tests, so no separate test is needed.